File: src/test/java/upbrella/be/config/interceptor/AuthTestConfig.java
package upbrella.be.config.interceptor;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import static org.springframework.core.Ordered.HIGHEST_PRECEDENCE;
import static org.springframework.core.Ordered.LOWEST_PRECEDENCE;

@Profile("test")
@Configuration
public class AuthTestConfig implements WebMvcConfigurer  {

    @Autowired
    private LoginInterceptor loginInterceptor;
    @Autowired
    private AdminInterceptor adminInterceptor;


    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(loginInterceptor)
                .order(HIGHEST_PRECEDENCE)
                .addPathPatterns("/**")
                .excludePathPatterns(
                        "/oauth/token",
                        "/user/me",
                        "/users/login/**",
                        "/users/oauth/login/**",
                        "/users/join/**",
                        "/stores/**",
                        "/index.html",
                        "/error/**",
                        "/api/error/**",
                        "/docs/**");


        registry.addInterceptor(adminInterceptor)
                .order(LOWEST_PRECEDENCE)
                .addPathPatterns("/admin/**")
                .excludePathPatterns(
                        "/oauth/token",
                        "/user/me",
                        "/users/login/**",
                        "/users/oauth/login/**",
                        "/users/join/**",
                        "/index.html",
                        "/error/**",
                        "/api/error/**",
                        "/docs/**");
    }
}

File: src/test/kotlin/upbrella/be/config/interceptor/AuthTestConfig.kt
package upbrella.be.config.interceptor

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

import org.springframework.core.Ordered.HIGHEST_PRECEDENCE
import org.springframework.core.Ordered.LOWEST_PRECEDENCE

@Profile("test")
@Configuration
class AuthTestConfig : WebMvcConfigurer {

    @Autowired
    private lateinit var loginInterceptor: LoginInterceptor

    @Autowired
    private lateinit var adminInterceptor: AdminInterceptor

    private val excludedPaths = listOf(
        "/oauth/token",
        "/user/me",
        "/users/login/**",
        "/users/oauth/login/**",
        "/users/join/**",
        "/stores/**",
        "/index.html",
        "/error/**",
        "/api/error/**",
        "/docs/**"
    )

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(loginInterceptor)
            .order(HIGHEST_PRECEDENCE)
            .addPathPatterns("/**")
            .excludePathPatterns(excludedPaths)

        registry.addInterceptor(adminInterceptor)
            .order(LOWEST_PRECEDENCE)
            .addPathPatterns("/admin/**")
            .excludePathPatterns(excludedPaths)
    }
}
